src/main/java/com/ericsson/ei/subscription/InformSubscriber.java

```diff
@@ -123,168 +131,223 @@ public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson)
+     * crumb was found.
      *
      * @param encoding
-     * @param notificationMeta
-     *     A String containing a URL
+     * @param notificationMeta A String containing a URL
      * @return JenkinsJsonCrumbData
+     * @throws AuthenticationException
      */
-    private JsonNode fetchJenkinsCrumbIfAny(String encoding, String notificationMeta) {
-        URL url;
+    private JsonNode fetchJenkinsCrumb(String encoding, String notificationMeta)
+            throws AuthenticationException {
         try {
-            String baseUrl = extractBaseUrl(notificationMeta);
-            url = new URL(baseUrl + JENKINS_CRUMB_ENDPOINT);
-        } catch (MalformedURLException e) {
-            LOGGER.error("Failed to format url to collect jenkins crumb");
-            return null;
-        }
+            URL url = buildJenkinsCrumbUrl(notificationMeta);
 
-        HttpHeaders headers = new HttpHeaders();
-        headers.add("Authorization", "Basic " + encoding);
-        headers.setContentType(MediaType.APPLICATION_JSON);
-        ResponseEntity<JsonNode> response = httpRequestSender.makeGetRequest(url.toString(), headers);
+            HttpHeaders headers = new HttpHeaders();
+            headers.add("Authorization", "Basic " + encoding);
+            headers.setContentType(MediaType.APPLICATION_JSON);
+            ResponseEntity<JsonNode> response = httpRequestSender.makeGetRequest(url.toString(),
+                    headers);
 
-        if (response == null || response.getStatusCodeValue() != HttpStatus.OK.value()) {
-            LOGGER.debug("No jenkins crumb found, most likely not jenkins or jenkins with crumb disabled");
-            return null;
-        }
+            JsonNode JenkinsJsonCrumbData = response.getBody();
+            return JenkinsJsonCrumbData;
 
-        LOGGER.debug("Successfully fetched Jenkins crumb.");
-        JsonNode JenkinsJsonCrumbData = response.getBody();
-        return JenkinsJsonCrumbData;
+        } catch (MalformedURLException e) {
+            String message = "Failed to format url to collect jenkins crumb.";
+            LOGGER.error(message, e);
+            throw new AuthenticationException(message, e);
+        } catch (HttpClientErrorException e) {
+            if (HttpStatus.UNAUTHORIZED == e.getStatusCode()) {
+                String message = "Failed to fetch crumb. Authentication failed, wrong username or password.";
+                LOGGER.error(message, e);
+                throw new AuthenticationException(message, e);
+            }
+            if (HttpStatus.NOT_FOUND == e.getStatusCode()) {
+                String message = String.format(
+                        "Failed to fetch crumb. The authentication type is %s,"
+                                + " but CSRF Protection seems disabled in Jenkins.",
+                        AUTHENTICATION_TYPE_BASIC_AUTH_JENKINS_CSRF);
+                LOGGER.warn(message, e);
+                return null;
```

Comment:
this return will cancel the coming throw. Should be so?

Reply:
Yes, to ensure that subscriptions will still trigger if CSRF security is off.